File: src/main/java/com/sstep/recipe/manager/dto/converter/RecipeDataConverter.java
package com.sstep.recipe.manager.dto.converter;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.sstep.recipe.manager.dto.RecipeData;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter(autoApply = true)
public class RecipeDataConverter implements AttributeConverter<RecipeData, String> {

    final ObjectMapper mapper = new ObjectMapper();

    @Override
    public String convertToDatabaseColumn(final RecipeData recipeData) {
        if (recipeData == null) {
            return null;
        }
        String value = null;
        try {
             value = mapper.writeValueAsString(recipeData);
        } catch (final JsonProcessingException e) {
            e.printStackTrace();
        }
        return value;
    }

    @Override
    public RecipeData convertToEntityAttribute(final String serializedData) {
        if (serializedData == null) {
            return null;
        }
        RecipeData recipeData = new RecipeData();
        try {
            recipeData = mapper.readValue(serializedData, RecipeData.class);
        } catch (final JsonProcessingException e) {
            e.printStackTrace();
        }
        return recipeData;
    }
}

File: src/main/kotlin/com/sstep/recipe/manager/dto/converter/RecipeDataConverter.kt
package com.sstep.recipe.manager.dto.converter

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.paramnames.ParameterNamesModule
import com.sstep.recipe.manager.dto.RecipeData
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter(autoApply = true)
class RecipeDataConverter : AttributeConverter<RecipeData?, String?> {

    override fun convertToDatabaseColumn(recipeData: RecipeData?): String? {
        recipeData ?: return null
        var value: String? = null
        try {
            value = mapper.writeValueAsString(recipeData)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
        }
        return value
    }

    override fun convertToEntityAttribute(serializedData: String?): RecipeData? {
        serializedData ?: return null
        var recipeData: RecipeData? = null
        try {
            recipeData = mapper.readValue(serializedData, RecipeData::class.java)
        } catch (e: JsonProcessingException) {
            e.printStackTrace()
        }
        return recipeData
    }

    private companion object {
        private val mapper = ObjectMapper().apply {
            registerModule(ParameterNamesModule())
        }
    }
}